SignalMessaging/environment/OWSSounds.m

```diff
@@ -245,6 +246,36 @@ + (OWSSound)globalNotificationSound
 
 + (void)setGlobalNotificationSound:(OWSSound)sound
 {
+    // Fallback push notifications play a sound specified by the server, but we don't want to store this configuration
+    // on the server. Instead, we create a file with the same name as the default to be played when receiving
+    // a fallback notification.
+    NSString *dirPath = [[OWSFileSystem appLibraryDirectoryPath] stringByAppendingPathComponent:@"Sounds"];
+    [OWSFileSystem ensureDirectoryExists:dirPath];
+
+    // This name is specified in the payload by the Signal Service when requesting fallback push notifications.
+    NSString *kDefaultNotificationSoundFilename = @"NewMessage.aifc";
+    NSString *defaultSoundPath = [dirPath stringByAppendingPathComponent:kDefaultNotificationSoundFilename];
+
+    DDLogDebug(@"%@ writing new default sound to %@", self.logTag, defaultSoundPath);
+
+    NSURL *_Nullable soundURL = [OWSSounds soundURLForSound:sound quiet:NO];
```

Comment:
The notification sounds are in different formats - they're not all .aifc files.  It seems like we should generate valid .aifc versions of all of the sounds if we're going to do this.  I can help - it shouldn't be hard.  i.e. we shouldn't count on iOS correctly playing sound files that have a misleading file extension.

Reply:
Oh, good catch!

> we shouldn't count on iOS correctly playing sound files that have a misleading file extension.

Totally agree.

Reply:
PTAL @charlesmchen - I converted the message tones to aifc via something like:

```
for file in *.m4r
do
ffmpeg  -i $file $(echo $file | sed 's/.m4r/.aifc')
done
```

And updated the -quite script accordingly.